tests/test_mock_temperature_sensor.py

```diff
@@ -0,0 +1,81 @@
+# This file is part of ts_ess_sensors.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+import asyncio
+import unittest
+
+from lsst.ts.ess.sensors.mock.mock_temperature_sensor import (
+    MockTemperatureSensor,
+    MIN_TEMP,
+    MAX_TEMP,
+)
+
+
+class MockTestCase(unittest.IsolatedAsyncioTestCase):
+    async def test_read_instrument(self):
+        num_channels = 4
+        name = "MockSensor"
+        ess_sensor = MockTemperatureSensor(name, num_channels)
+        ess_sensor.terminator = "\r\n"
+        loop = asyncio.get_event_loop()
+        device_name, err, resp = await loop.run_in_executor(None, ess_sensor.readline)
+        self.assertEqual(name, device_name)
+        resp = resp.strip(ess_sensor.terminator)
+        data = resp.split(",")
+        for i in range(0, num_channels):
```

Comment:
Do you expect the length of data to match num_channels? If so, please test for that before this loop. It will avoid confusing error messages if there's not enough data. (That's a good idea even if it's OK to have *more* than num_channels items, though I hope they'll match because you ignore any additional data items).

Reply:
Good idea. Done.